hybrid/pkg/agent/agent_test.go

```diff
@@ -0,0 +1,386 @@
+var payloadOneData = `{"cluster":"hybrid-node-attestor_fake","token":"part1.part2.part3-part4-part5--part6-part7"}`
+var payloadTwoData = `{"document":"{\n  \"accountId\" : \"123456789_TEST\",\n  \"architecture\" : \"x86_64\",\n  \"availabilityZone\" : \"us-east-2a\",\n  \"billingProducts\" : null,\n  \"devpayProductCodes\" : null,\n  \"marketplaceProductCodes\" : null,\n  \"imageId\" : \"ami-010203040506\",\n  \"instanceId\" : \"i-010203040506\",\n  \"instanceType\" : \"m5.large\",\n  \"kernelId\" : null,\n  \"pendingTime\" : \"2022-09-22T03:22:21Z\",\n  \"privateIp\" : \"192.168.77.116\",\n  \"ramdiskId\" : null,\n  \"region\" : \"us-east-2\",\n  \"version\" : \"2017-09-30\"\n}","signature":"eO4+90PuN8bZaIJjpBe1/mAzPhvSrrhLATwPFaOPzK5ZSUpsbVOuK2tXjMYkx+ora7mcaL0G45li\nbZLGUIee+DF/YZ8/5RuNf1Z8yn+5e2AqLvNhIsF5IOVZWk8yDvl/jBJCcW8GaRblldWdMoDiC2OA\nqVyRjyJCXUySNu0JADE="}`
+
+// ------------------------------------------------------------------------------------------------------------------------
+
+func TestNew(t *testing.T) {
+	hybridPlugin := New()
+	require.NotNil(t, hybridPlugin, "New should return a non-nil value")
+	require.IsType(t, &HybridPluginAgent{}, hybridPlugin, "New should return a HybridPluginAgent")
+}
+
+func TestMethodsThatParseHclConfig(t *testing.T) {
+	interceptor := new(InterceptorWrapper)
+	plugin := HybridPluginAgent{interceptor: interceptor}
+
+	pluginAstNode, err := plugin.decodeStringAndTransformToAstNode(pluginsString)
+
+	require.NoError(t, err, "Error decoding test string")
+	require.Len(t, pluginAstNode, 2, "Could not transform HCL string configuration: %w", err)
+	require.Contains(t, pluginAstNode, "k8s_psat", "Could not access k8s_psat plugin by index on ast node")
+	require.Contains(t, pluginAstNode, "aws_iid", "Could not access aws_iid plugin by index on ast node")
+
+	pluginNames, pluginsData := plugin.parseReceivedData(pluginAstNode)
+
+	require.Len(t, pluginNames, 2, "Could not parse plugin names")
+	require.Contains(t, pluginNames, "k8s_psat", "Could not access k8s_psat plugin by index after parsing")
+	require.Contains(t, pluginNames, "aws_iid", "Could not access aws_iid plugin by index after parsing")
+
+	require.Len(t, pluginsData, 2, "Could not parse plugin data")
+	require.Contains(t, pluginsData, "k8s_psat", "Could not access k8s_psat plugin by index after parsing")
+	require.Equal(t, "\n  cluster = \"hybrid-node-attestor\"\n", pluginsData["k8s_psat"], "k8s_psat plugin data was not extracted properly")
+	require.Contains(t, pluginsData, "aws_iid", "Could not access aws_iid plugin by index after parsing")
+	require.Equal(t, "\n  accountId = 728109058939\n", pluginsData["aws_iid"], "aws_iid plugin data was not extracted properly")
+}
+
+func TestSupportedPluginsInitialization(t *testing.T) {
+	interceptor := new(InterceptorWrapper)
+	plugin := HybridPluginAgent{interceptor: interceptor}
+
+	plugins, err := plugin.initPlugins([]string{"aws_iid", "k8s_psat", "azure_msi", "gcp_iit"})
```

Comment:
are these all the supported plugins? if not, we should add them all here

Reply:
yes, they're all plugins supported right now